pkg/scalers/splunk_o11y_scaler.go

```diff
@@ -0,0 +1,264 @@
+		}
+		meta.queryValue = queryValue
+	} else {
+		if config.AsMetricSource {
+			meta.queryValue = 0
+		} else {
+			return nil, fmt.Errorf("no queryValue given")
+		}
+	}
+
+	// activationQueryValue
+	meta.activationQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
+		activationQueryValue, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, fmt.Errorf("queryValue parsing error %w", err)
+		}
+		meta.activationQueryValue = activationQueryValue
+	}
+
+	// queryAggregator
+	if val, ok := config.TriggerMetadata["queryAggregator"]; ok && val != "" {
+		queryAggregator := strings.ToLower(val)
+		switch queryAggregator {
+		case "max", "min", "avg":
+			meta.queryAggegrator = queryAggregator
+		default:
+			return nil, fmt.Errorf("queryAggregator value %s has to be one of 'max', 'min', or 'avg'.", queryAggregator)
+		}
+	} else {
+		meta.queryAggegrator = ""
+	}
+
+	// accessToken
+	if accessToken, ok := config.AuthParams["accessToken"]; ok {
+		// cleanedAccessToken := strings.ReplaceAll(accessToken, "\n", "")
+		// cleanedAccessToken = strings.ReplaceAll(accessToken, "\r", "")
+		meta.accessToken = accessToken
+	} else {
+		return nil, fmt.Errorf("no accessToken given")
+	}
+
+	// realm
+	if realm, ok := config.TriggerMetadata["realm"]; ok {
+		meta.realm = realm
+	} else {
+		return nil, fmt.Errorf("no realm given")
+	}
+	logger.Info("Splunk Realm -> %s\n", meta.realm)
+
+	// Debug TODO check
+	meta.vType = v2.ValueMetricType
+
+	return &meta, nil
+}
+
+func newSplunkO11yConnection(ctx context.Context, meta *splunkO11yMetadata, config *scalersconfig.ScalerConfig) (*signalflow.Client, error) {
+	accessToken := meta.accessToken
+	realm := meta.realm
+
+	if realm == "" || accessToken == "" {
+		return nil, fmt.Errorf("error. could not find splunk access token or ream.")
+	}
+
+	apiClient, err := signalflow.NewClient(
+		signalflow.StreamURLForRealm(realm),
+		signalflow.AccessToken(accessToken))
+	if err != nil {
+		return nil, fmt.Errorf("error creating SignalFlow client: %w", err)
+	}
+
+	return apiClient, nil
+}
+
+func logMessage(logger logr.Logger, msg string, value float64) {
+	if value != -1 {
+		msg = fmt.Sprintf("splunk_o11y_scaler: %s -> %v", msg, value)
+	} else {
+		msg = fmt.Sprintf("splunk_o11y_scaler: %s", msg)
+	}
+	logger.Info(msg)
+}
+
+func (s *splunkO11yScaler) getQueryResult(ctx context.Context) (float64, error) {
+	// var duration time.Duration = 1000000000 // one second in nano seconds
+	var duration time.Duration = 10000000000 // ten seconds in nano seconds
+
+	comp, err := s.apiClient.Execute(context.Background(), &signalflow.ExecuteRequest{
+		Program: s.metadata.query,
+	})
+	if err != nil {
+		return -1, fmt.Errorf("error: could not execute signalflow query: %w", err)
+	}
+
+	go func() {
+		time.Sleep(duration)
+		if err := comp.Stop(context.Background()); err != nil {
+			s.logger.Info("Failed to stop computation")
+		}
+	}()
```

Comment:
You should rather use [`WithTimeout()`](https://pkg.go.dev/context#WithTimeout):
```suggestion
	var duration time.Duration = 10000000000 // ten seconds in nano seconds

        xCtx, cancel := context.WithTimeout(ctx, duration)
        defer cancel()
	comp, err := s.apiClient.Execute(xCtx, &signalflow.ExecuteRequest{
		Program: s.metadata.query,
	})
	if err != nil {
		return 0.0, fmt.Errorf("error: could not execute signalflow query: %w", err)
	}
```

Reply:
Yes, that makes sense. I will update the function accordingly.